cmake/OpenCVFindProtobuf.cmake

```diff
@@ -67,22 +74,51 @@ else()
   endif()
 endif()
 
+# See https://github.com/opencv/opencv/issues/24369
+# In Protocol Buffers v22.0 and later drops C++11 support and depends abseil-cpp.
+#   Details: https://protobuf.dev/news/2022-08-03/
+# And if std::text_view is in abseil-cpp requests C++17 and later.
+
+if(HAVE_PROTOBUF)
+    if("${Protobuf_VERSION}" MATCHES [[[0-9]+.([0-9]+).[0-9]+]])
+        string(COMPARE GREATER "${CMAKE_MATCH_1}" "21" REQUEST_AFTER_CXX17)  # >=22
+
+        if(REQUEST_AFTER_CXX17)
+            string(COMPARE GREATER "${CMAKE_CXX_STANDARD}" "16" USED_AFTER_CXX17)  # >=17
+
+            if(NOT USED_AFTER_CXX17)
+                message("CMAKE_CXX_STANDARD : ${CMAKE_CXX_STANDARD}")
+                message("protobuf           : ${Protobuf_VERSION}")
+                message(FATAL_ERROR "protobuf(v22 and later) and abseil-cpp request CMAKE_CXX_STANDARD=17 and later.")
```

Comment:
Usually OpenCV does not throw errors if an optional dependency can not be found or used. @asmorkalov , @opencv-alalek , what do you think? Should we show a warning and disable protobuf?

Reply:
Thank you for your review !
I would like to wait for comments for the direction of correction.

And unfortunately, there are some `message(FATAL_ERROR)` in this file like this block. If error is occured, they will stop configuration.

https://github.com/opencv/opencv/blob/5199850039ad23f1f0e6cccea5061a9fea5efca6/modules/dnn/CMakeLists.txt#L133-L135

https://github.com/opencv/opencv/blob/5199850039ad23f1f0e6cccea5061a9fea5efca6/modules/dnn/CMakeLists.txt#L166-L168

https://github.com/opencv/opencv/blob/5199850039ad23f1f0e6cccea5061a9fea5efca6/modules/dnn/CMakeLists.txt#L228-L230

Reply:
It is not a default configuration.
Need to explicitly specify `BUILD_PROTOBUF=OFF` to go here.
I believe we could these errors here.

Reply:
Thank you for your review!
Yes, it is clear. This is about `protobuf`, so I set HAVE_PROTOBUF=OFF if old cxx standard is used. (And I forget C++98 ...)